File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.java
class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
        int[] a = new int[nums.length];
        Set<Integer> s = new HashSet<>();
        for(int i = 0; i < nums.length; i++) {
            int r = nums[i] % k;
            if(i == 1) {
                s.add(0);
            }
            else if(i > 1) {
                s.add(a[i - 2]);
            }
            if(i != 0) {
                r = (r + a[i - 1]) % k;
            }
            if(s.contains(r))
                return true;
            a[i] = r;
        }
        return false;
    }
} 

File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
        set<int> s = {};
        for(int i = 0; i < nums.size(); i++) {
            nums[i] %= k;
            if(i != 0) {
                nums[i] = (nums[i] + nums[i - 1]) % k;
            }
            if(i == 1) {
                s.insert(0);
            }
            else if(i > 1) {
                s.insert(nums[i - 2]);
            }
            
            if(bool(s.count(nums[i])))
                return true;
        }
        return false;
    }
};